model/execution/ProctoredDeliveryFactoryEventsService.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA ;
+ *
+ * @author Aleksej Tikhanovich <aleksej@taotedting.com>
+ */
+
+namespace oat\taoProctoring\model\execution;
+
+use oat\generis\model\data\ModelManager;
+use oat\oatbox\event\Event;
+use oat\oatbox\service\ConfigurableService;
+use oat\oatbox\service\ServiceManager;
+use oat\taoProctoring\model\ProctorService;
+
+class ProctoredDeliveryFactoryEventsService extends ConfigurableService

```

Comment:
In my opinion this is not EventsService. This is event handler. I believe we do not need to have this service. You could add this option and event handlers to the existing `oat\taoProctoring\model\ProctorService` service.